gitpod-io/gitpod
# [jb-remote] hook gp cli

## Description


This PR enables support of gp CLI in JB products, namely credential helper, open and preview commands.
Caveat: It breaks gp cli in VS Code running alongside of JB products. We will have a look into separately: https://gitpod.slack.com/archives/C01KGM9BH54/p1644223793552719

This PR also include project specific IntelliIJ configs to enable dog fooding.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/7980, fixes https://github.com/gitpod-io/gitpod/issues/7848

## How to test


- Start a workspace with IntelliJ: https://ak-jb-remote-git.staging.gitpod-dev.com/#referrer:jetbrains-gateway/https://github.com/gitpod-io/spring-petclinic
- Try git push from cli and UI, you should see a notification about missing scopes in both cases. Clicking on the link should take you in access control page.
  - Sometimes UI is buggy and git integration is stuck, try to reopen then. It is not related to changes in this PR.
- Grand missing scopes in the access control page and check that you can now trigger push from cli and UI without issues.
- `gp open pom.xml` should open pom.xml file from terminal
- `gp open pom.xml --wait` should open pom.xml file from terminal and block it till file is not closed
- `gp preview $(gp url 8080)` should open the browser page

## Design Review

You can find pair review with @felladrin [here](https://drive.google.com/file/d/1qh7eaGOsmHa5Jb2gag3oEcLq8JX5QBbQ/view?usp=sharing).

![gp-idea drawio](https://user-images.githubusercontent.com/3082655/153171143-7a944592-f805-4fce-a8c9-72cbceacb456.png)

## Release Notes

```release-note
NONE
```

## Documentation
